code/game/objects/items/weapons/implant/implants/carrion/holo_spider.dm

```diff
@@ -0,0 +1,237 @@
+/obj/item/implant/carrion_spider/holographic
+	name = "holographic spider"
+	desc = "A spider with a peculiarly reflective surface"
+	icon_state = "spiderling_holographic"
+	slot_flags = SLOT_ID | SLOT_BELT | SLOT_EARS | SLOT_HOLSTER | SLOT_BACK | SLOT_MASK | SLOT_GLOVES | SLOT_HEAD | SLOT_OCLOTHING | SLOT_ICLOTHING | SLOT_FEET | SLOT_EYES
+	spider_price = 5
+	var/can_use = 1
+	var/saved_name
+	var/saved_description
+	var/saved_item
+	var/saved_type
+	var/saved_icon
+	var/saved_icon_state
+	var/saved_layer
+	var/saved_original_plane
+	var/saved_dir
+	var/saved_message
+	var/saved_appearance
+	var/saved_item_state
+	var/saved_w_class
+	var/spider_appearance
+	var/saved_gender
+
+	var/dummy_active = FALSE
+	var/scan_mobs = TRUE
+
+
+/obj/item/implant/carrion_spider/holographic/activate()
+	..()
+	toggle()
+
+/obj/item/implant/carrion_spider/holographic/toggle_attack(mob/user)
+	if(ready_to_attack)
+		ready_to_attack = FALSE
+		to_chat(user, SPAN_NOTICE("\The [src] wont attack nearby creatures anymore and can be used to scan creatures without attaching itself to them."))
+		scan_mobs = TRUE
+	else
+		ready_to_attack = TRUE
+		to_chat(user, SPAN_NOTICE("\The [src] is ready to attack nearby creatures or to be attached manually"))
+		scan_mobs = FALSE
+
+/obj/item/implant/carrion_spider/holographic/attack(mob/living/M, mob/living/user)
+	if(scan_mobs)
+		return
+	else
+		..()
+
+/obj/item/implant/carrion_spider/holographic/afterattack(atom/target, mob/user, proximity)
+	if(istype(target, /obj/item/storage)) 
+		return
+	if(!proximity)
+		return
+	if(dummy_active)
+		return
+	if(!scan_mobs)
+		return
+	reset_data()
+	playsound(get_turf(src), 'sound/weapons/flash.ogg', 100, 1, -6)
+	to_chat(user, SPAN_NOTICE("Scanned [target]."))
+	saved_name = target.name
+	saved_item = target
+	saved_type = target.type 
+	saved_icon = target.icon
+	saved_icon_state = target.icon_state
+	saved_description = target.desc
+	saved_dir = target.dir
+	saved_appearance = target.appearance
+	saved_gender = target.gender
+	spider_appearance = src.appearance
+	saved_layer = target.layer
+	saved_original_plane = target.original_plane
+	if(istype(target, /obj))	
+		var/obj/O = new saved_type(src)
+		saved_item_state = O.item_state
+		saved_w_class = O.w_class
+		qdel(O)
```

Comment:
Why instance a new object here, instead of using the object that is being scanned?

Reply:
adressed